Caltech-IPAC/firefly
# DM-8668 lc period finder

The development includes:
- period finding layout before and after periodogram which contains the panels 
   (LcPeriods.jsx)
    . to enter values of period finding parameters
    . to show phase folding preview of xy plot
    . to show periodogram and peak tables and XT plots (after periodogram)
- Find periodogram dialog  which inherits some parameters from period finding panel
   (LcPeriodogram.jsx)
- Revamp LcManager to work with period finding and periodogram finding flow 
   (LcManager.js)
- add 'keepMounted' property to FieldGroup component which allows the parameters under the field group to be kept mounted while the component is unmounted. 
- update 'RangeSlider' component by removing the text input field attached to the slider. 
Test:
- localhost:8080/firefly/lc.html. 
- choose file - lc_raw.tbl ->search, 'Periof Finding' button is shown on the top menu after the search. 
- click button 'Period Finding' to get period finding layout
- enter values or move the slider in period finding panel to see the change of phase folding XY chart and the change of the buttons at the bottom ( 'Back to Period', 'Accept Period') 
- click button 'Find Periodogram' to get Periodogram popup dialog
   . 'reset' button restores the values to those as the popup is started. 
   . 'Periodogram Calculation' button exits the dialogs and calculates periodogram/peak tables and charts. 
  . 'cancel' button restores the field values and exits the dialog. 
- select the period from the periodogram/peak tables or charts to see the change of phase folding  plot.
